File: cpp/problem25.cpp
#include <stdio.h>
#include <stdlib.h>

using namespace std;

#include "bignum.h"

#define uint unsigned int

int main(int argc, char* argv[])
{
	Bignum p = 1;
	Bignum q = 1;
	Bignum t;
	uint i = 2;
	uint length;
	uint limit = 1000;
	char buffer[1200];

	while (true)
	{
		t = p + q;
		p = q;
		q = t;
		
		i++;

		length = t.str(buffer, 1200) - 1;
		
		if (length >= limit)
		{
			printf("%u", i);
			break;
		}

	}

	printf("\n");
	return 0;
}

File: python/problem25.py

def fib():
    p = 1
    q = 1

    while True:
        yield p

        p, q = q, p + q

limit = 1000
i = 1

for n in fib():
    s = str(n)

    if len(s) >= limit:
        print i
        break

    i += 1
